cmd/probe/internal/component/postgres/apecloudpostgres/manager.go

```diff
@@ -254,34 +257,57 @@ func (mgr *Manager) IsCurrentMemberHealthy(ctx context.Context, cluster *dcs.Clu
 // IsMemberHealthy firstly get the leader's connection pool,
 // because only leader can get the cluster healthy view
 func (mgr *Manager) IsMemberHealthy(ctx context.Context, cluster *dcs.Cluster, member *dcs.Member) bool {
-	IPPort := mgr.Config.GetConsensusIPPort(cluster, member.Name)
-
-	sql := fmt.Sprintf(`select connected, log_delay_num from consensus_cluster_health where ip_port = '%s';`, IPPort)
-	resp, err := mgr.QueryLeader(ctx, sql, cluster)
+	healthStatus, err := mgr.getMemberHealthStatus(ctx, cluster, member)
 	if errors.Is(err, postgres.ClusterHasNoLeader) {
 		mgr.Logger.Infof("cluster has no leader, will compete the leader lock")
 		return true
 	} else if err != nil {
-		mgr.Logger.Errorf("query %s with leader failed, err:%v", sql, err)
+		mgr.Logger.Errorf("check member healthy failed, err:%v", err)
 		return false
 	}
 
+	return healthStatus.Connected
+}
+
+func (mgr *Manager) getMemberHealthStatus(ctx context.Context, cluster *dcs.Cluster, member *dcs.Member) (*postgres.ConsensusMemberHealthStatus, error) {
+	if mgr.DBState != nil && mgr.healthStatus != nil {
+		return mgr.healthStatus, nil
+	}
```

Comment:
the healthStatus only requests once and then keep using caching?

Reply:
healthStatus will be used in IsMemberHealthy and IsMemberLagging, so cache it.

Reply:
GetDBState will get most required database kernel states of current member in one HA loop.